test/linarith.lean

```diff
@@ -431,13 +431,11 @@ example (f : ℤ → E) (h : 0 = f 0) : 1 ≤ 2 := by nlinarith
 example (a : E) (h : a = a) : 1 ≤ 2 := by nlinarith
 end
 
+-- This should not be slower than the example below with the Fourier-Motzkin oracle
```

Comment:
I've pushed some improvements. The results I see using `set_option trace.profiler true` aren't consistent, but the sparse simplex is always a bit faster than FM on my machine (2.33s vs. 2.55s). I've measured the time spent by the oracles specifically: the sparse simplex took 79ms (57ms on Gauss Elimination and 15ms on the Simplex Algorithm), while the FM oracle took 110ms.

Reply:
Actually, I don't understand why Gauss Elimination (and the oracle in general) is so slow. I am new to programming in Lean and don't know how fast it should be. In C, the GE method would take less than 1ms on a matrix of this size (100x100).

Reply:
This is strange. I'm also seeing somewhat inconsistent timing, but even with your new changes the order is pretty consistently 
> FM seems to outperform dense simplex which outperforms spare simplex

Even weirder: explicitly providing `simplexAlgorithmSparse` as the oracle seems to be 100ms faster than letting it be found as the default (~750ms vs 850ms on my laptop)! I have no idea why.

Reply:
> Even weirder: explicitly providing `simplexAlgorithmSparse` as the oracle seems to be 100ms faster than letting it be found as the default (~750ms vs 850ms on my laptop)! I have no idea why.

On my machine, I don't experience this. It seems like all of this is highly dependent on the hardware.

Reply:
Oh, yes, the simplex oracle is definitely much faster than FM on large states.

It seems like some of my weird timing measurements were artifacts of my testing setup. Sorry for the noise. I'm now seeing the same behavior you are, sparse simplex is about 10% faster than FM on this example.

Reply:
I agree with you that I'd expect the GE step to be faster. These things are notoriously tricky to get right in a functional language. Since this is already an improvement, I'm willing to take this PR as-is and deal with speeding up the implementation in future work!

Reply:
Could you add the large state example (the one for which FM takes over a minute) to the test file, with a comment that it's too big for FM but accessible to simplex?

Reply:
> Could you add the large state example (the one for which FM takes over a minute) to the test file, with a comment that it's too big for FM but accessible to simplex?

Done :)